contxt/auth/cli.py

```diff
@@ -1,59 +1,196 @@
 from getpass import getpass
+from json import dump, load
+from pathlib import Path
+from typing import Dict, Optional
 
-from contxt.auth import BaseAuth
-from contxt.services.auth import ContxtAuthService
-from contxt.utils import Config, make_logger
+from auth0.v3.authentication import GetToken
+
+from contxt.auth import BaseAuth, DependentTokenProvider, TokenProvider
+from contxt.services.auth import AuthService
+from contxt.utils import make_logger
 
 logger = make_logger(__name__)
 
 
-class CLIAuth(BaseAuth):
+class Auth0TokenProvider(TokenProvider):
```

Comment:
just naming question: IIUC, all tokens are ultimately provided by Auth0 services, so wonder if we can rename the classes to something more mnemonic, like following the use pattern of: `UserTokenProvider` vs `MachineTokenProvider`

Reply:
yea i'm bad with naming and couldn't figure out what to call this haha